### PR TITLE
fixing problem where initial conditions are reperturbed

### LINKED ISSUE #164: 🐛[BUG]: Running inference_ensemble with multiple ensemble members per rank causes perturbations to be applied multiple times
### Version

source - main @ git commit: ebd86637ec853c11e9ff5e717b053351c740e526

### On which installation method(s) does this occur?

Source

### Describe the issue

When running 32 ensemble members with 1 rank, the later ensemble members have the perturbation applied multiple times.   So the second ensemble member on each rank has a perturbation applied twice.  The third ensemble member has a perturbation applied three times.  Etc.   Is this behavior expected?

This could make the magnitude of the perturbations very large, unless there is some perturbation magnitude normalization applied later.

Using the correlated IC perturbation method with noise amplitude 0.05, the first ensemble member has this magnitude of perturbation: 
image

The 15th ensemble member has this perturbation:
image

And the 31st ensemble member has this perturbation:
image

I think there could be two causes:

1. x is being overwritten by the perturb method [here](https://github.com/NVIDIA/earth2mip/blob/main/earth2mip/inference_ensemble.py#L104), so as the loop moves forward, the perturbed x gets perturbed again, instead of the original x getting perturbed.
2. x is mutable, and x is being modified [here](https://github.com/NVIDIA/earth2mip/blob/main/earth2mip/inference_ensemble.py#L256).  

### Environment details

_No response_

### PR DESCRIPTION
**Pull request by ankurmahesh**
# Earth-2 MIP Pull Request

## Description
Fixes #164 

This pull request changes `earth2mip/inference_ensemble` to avoid x from getting reperturbed during each loop's rank over ensemble members.  Previously, x would be perturbed, and then the perturbed x would get perturbed again, and then the perturbed x would get perturbed again.  This could make the magnitude of perturbations very large if many ensemble members are run per rank.

This MR prevents x from being overwritten during the loop over ensemble members.
## Checklist

- [ x] I am familiar with the Contributing Guidelines.
- [ ] New or existing tests cover these changes.
- [ x] The documentation is up to date with these changes.
- [ x] The CHANGELOG.md is up to date with these changes.
- [ x] An [issue](https://github.com/NVIDIA/earth2mip/issues) is linked to this pull request.

### CODE REVIEW

File: earth2mip/inference_ensemble.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def run_ensembles(
         batch_size = min(batch_size, n_ensemble - batch_id)
 
         x = x.repeat(batch_size, 1, 1, 1, 1)
```
Comment (yairchn) [inline earth2mip/inference_ensemble.py:103]: @ankurmahesh 
@NickGeneva 
this in place operation is repeating x needlessly many times, right? this can grew exponentially I think we can think of better way to code this irrespectively of the perturbation part.

Comment (nbren12) [reply earth2mip/inference_ensemble.py:103]: This PR doesn't touch this code. Wouldn't we have noticed this before?

**Review by nbren12 — COMMENTED** [review]
LGTM. Nice simple fix avoiding `+=`.

Can you add a unit test  [here](https://github.com/NVIDIA/earth2mip/blob/main/test/test_ensemble_perturb.py) which proves the problem goes away. Maybe you can call `perturb` and assert that the input is unmodified?

Would be nice to have better test coverage of this ensemble perturbations.

**Comment by nbren12** [comment]
just a btw, but I didn't understand that += modified the actually contents of a torch or numpy array rather than making a new array and reusing the name "x". Seems this is the typical behavior for `+=` on stateful objects.

```
[ins] In [5]: x = torch.tensor([2.0])

[ins] In [6]: y = x = torch.tensor([2.0])

[ins] In [7]: x +=1

[ins] In [8]: y
Out[8]: tensor([3.])

[ins] In [9]: import numpy as np

[ins] In [10]: x = np.array([2.0])

[ins] In [11]: y = x

[ins] In [12]: x +=1

[ins] In [13]: y
Out[13]: array([3.])

[ins] In [14]: x = y = []

[ins] In [15]: x += [1]

[ins] In [16]: y
Out[16]: [1]

[ins] In [17]: x =y =1

[ins] In [18]: x += 1

[ins] In [19]: y
Out[19]: 1
```
